pycbc/conversions.py

```diff
@@ -764,23 +764,236 @@ def snr_from_loglr(loglr):
+        newshape = origshape
+    f0s = numpy.zeros((mass.size, nmodes))
+    taus = numpy.zeros((mass.size, nmodes))
+    mass = mass.ravel()
+    spin = spin.ravel()
+    l = l.ravel()
+    m = m.ravel()
+    qnmfreq = None
+    modes = range(nmodes)
+    for ii in range(mass.size):
+        qnmfreq = _genqnmfreq(mass[ii], spin[ii], l[ii], m[ii], nmodes,
+                              qnmfreq=qnmfreq)
+        f0s[ii, :] = [qnmfreq.data[n].real/(2 * numpy.pi) for n in modes]
+        taus[ii, :] = [1./qnmfreq.data[n].imag for n in modes]
+    f0s = f0s.reshape(newshape)
+    taus = taus.reshape(newshape)
+    return (formatreturn(f0s, input_is_array),
+            formatreturn(taus, input_is_array))
+
+
+def freq_from_final_mass_spin(final_mass, final_spin, l=2, m=2, nmodes=1):
+    """Returns QNM frequency for the given mass and spin and mode.
 
-def freq_from_final_mass_spin(final_mass, final_spin, l=2, m=2):
-    return get_lm_f0tau(final_mass, final_spin, l, m, 1)[0][0]
+    Parameters
+    ----------
+    final_mass : float or array
+        Mass of the black hole (in solar masses).
+    final_spin : float or array
+        Dimensionless spin of the final black hole.
+    l : int or array, optional
+        l-index of the harmonic. Default is 2.
+    m : int or array, optional
+        m-index of the harmonic. Default is 2.
+    nmodes : int, optional
+        The number of overtones to generate. Default is 1.
 
+    Returns
+    -------
+    float or array
+        The frequency of the QNM(s), in Hz. If only a single mode is requested
+        (and mass, spin, l, and m are not arrays), this will be a float. If
+        multiple modes requested, will be an array with shape
+        ``[input shape x] nmodes``, where ``input shape`` is the broadcasted
+        shape of the inputs.
+    """
+    return get_lm_f0tau(final_mass, final_spin, l, m, nmodes)[0]
+
+
+def tau_from_final_mass_spin(final_mass, final_spin, l=2, m=2, nmodes=1):
+    """Returns QNM damping time for the given mass and spin and mode.
+
+    Parameters
+    ----------
+    final_mass : float or array
+        Mass of the black hole (in solar masses).
+    final_spin : float or array
+        Dimensionless spin of the final black hole.
+    l : int or array, optional
+        l-index of the harmonic. Default is 2.
+    m : int or array, optional
+        m-index of the harmonic. Default is 2.
+    nmodes : int, optional
+        The number of overtones to generate. Default is 1.
+
+    Returns
+    -------
+    float or array
+        The damping time of the QNM(s), in seconds. If only a single mode is
+        requested (and mass, spin, l, and m are not arrays), this will be a
+        float. If multiple modes requested, will be an array with shape
+        ``[input shape x] nmodes``, where ``input shape`` is the broadcasted
+        shape of the inputs.
+    """
+    return get_lm_f0tau(final_mass, final_spin, l, m, nmodes)[1]
+
+
+# following are from Berti et al. 2006.
```

Comment:
Could we put the arxiv number here? Berti et al. have a few papers from 2006 and it would be easier to find which one (especially if we want to extend this to higher order modes some day).

Reply:
@micamu Ok, I added the reference.